test/jdk/java/awt/Window/TestAppFreeze.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+import java.awt.BorderLayout;
+import java.awt.FlowLayout;
+import java.awt.event.ActionEvent;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ *  @test
+ *  @bug 8256465
+ *  @requires (os.family == "Mac")
+ *  @summary Application should not freeze when opening dialog
+ *  @key headful
+ *  @run main/manual TestAppFreeze
+ */
+
+public class TestAppFreeze {
+    private static volatile CountDownLatch countDownLatch;
+    private static JFrame instructionFrame;
+    private static JFrame testFrame;
+    private static volatile boolean testPassed = false;
+
+    private static boolean validatePlatform() {
+        String osName = System.getProperty("os.name");
+        if (osName == null) {
+            throw new RuntimeException("Name of the current OS could not be" +
+                    " retrieved.");
+        }
+        return osName.startsWith("Mac");
+    }
+
+    private static void createInstructionUI() {
+        SwingUtilities.invokeLater(() -> {
+            String instruction = "1. This test is only for Mac OS Version 11 " +
```

Comment:
Yeah, i think it will affect Mac OS 10.12.+  as the tabbedWindows was introduced in 10.12 , i have checked it affects the 10.14 , don't have earlier version to check.